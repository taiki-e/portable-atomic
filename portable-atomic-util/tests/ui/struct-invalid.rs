pub mod struct_no_field {
    use portable_atomic::Atomicable;

    #[derive(Clone, Copy, Atomicable)]
    struct A {}
}

pub mod struct_no_repr {
    use portable_atomic::Atomicable;

    #[derive(Clone, Copy, Atomicable)]
    struct A {
        f0: u8,
        f1: u8,
    }
}

fn main() {}
