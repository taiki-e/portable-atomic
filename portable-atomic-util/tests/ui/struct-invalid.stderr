error: #[derive(Atomicable)] may not be used on struct with no field
 --> tests/ui/struct-invalid.rs:5:12
  |
5 |     struct A {}
  |            ^

error: repr must be must specified
  --> tests/ui/struct-invalid.rs:12:12
   |
12 |     struct A {
   |            ^
